packages/openactive-broker-microservice/src/sample-requests.js

```diff
@@ -0,0 +1,157 @@
+  telephone: '020 811 8055',
+  givenName: 'Geoff',
+  familyName: 'Capes',
+};
+
+function checkpointOne(opportunity, seller, offer) {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'OrderQuote',
+    brokerRole: 'https://openactive.io/AgentBroker',
+    broker,
+    seller: seller['@id'],
+    orderedItem: [
+      {
+        '@type': 'OrderItem',
+        position: 0,
+        acceptedOffer: offer['@id'],
+        orderedItem: opportunity['@id'],
+      },
+    ],
+  };
+}
+
+function checkpointTwo(opportunity, seller, offer) {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'OrderQuote',
+    brokerRole: 'https://openactive.io/AgentBroker',
+    broker,
+    seller: seller['@id'],
+    customer,
+    orderedItem: [
+      {
+        '@type': 'OrderItem',
+        position: 0,
+        acceptedOffer: offer['@id'],
+        orderedItem: opportunity['@id'],
+      },
+    ],
+  };
+}
+
+function book(opportunity, seller, offer) {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'Order',
+    brokerRole: 'https://openactive.io/AgentBroker',
+    broker,
+    seller: seller['@id'],
+    customer,
+    orderedItem: [
+      {
+        '@type': 'OrderItem',
+        position: 0,
+        acceptedOffer: offer['@id'],
+        orderedItem: opportunity['@id'],
+      },
+    ],
+  };
+}
+
+function proposal(opportunity, seller, offer) {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'OrderProposal',
+    brokerRole: 'https://openactive.io/AgentBroker',
+    broker,
+    seller: seller['@id'],
+    customer,
+    orderedItem: [
+      {
+        '@type': 'OrderItem',
+        position: 0,
+        acceptedOffer: offer['@id'],
+        orderedItem: opportunity['@id'],
+      },
+    ],
+  };
+}
+
+function proposalUpdate() {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'OrderProposal',
+    orderProposalStatus: 'https://openactive.io/CustomerRejected',
+    orderCustomerNote: "Sorry I've actually made other plans, hope you find someone!",
+  };
+}
+
+function orderCancellation() {
+  return {
+    '@context': 'https://openactive.io/',
+    '@type': 'Order',
+    orderedItem: [
+      {
+        '@type': 'OrderItem',
+        '@id': '',
+        orderItemStatus: 'https://openactive.io/CustomerCancelled',
+      },
+    ],
+  };
+}
+
+// NOTE: duplicated from openactive-integration-tests/test/helpers/flow-stages/fetch-opportunities.js
+function getRandomRelevantOffer(opportunity, criteriaName) {
+  const relevantOffers = getRelevantOffers(criteriaName, opportunity, {
+    harvestStartTime: HARVEST_START_TIME,
+  });
+  if (relevantOffers.length === 0) { return null; }
+
+  return relevantOffers[Math.floor(Math.random() * relevantOffers.length)];
+}
+
+function buildSampleRequests(opportunity, criteriaName) {
+  const parent = opportunity.superEvent || opportunity.facilityUse;
+  const seller = parent.organizer || parent.provider;
```

Comment:
For the request this should be the `@id` rather than the whole seller object, so can use this too:

https://github.com/openactive/openactive-test-suite/blob/6f477cfc64f318ba198d93c5f83db31daa38d1cf/packages/openactive-broker-microservice/app.js#L943-L950
```suggestion
  const sellerId = detectSellerId(opportunity);
```